### PR TITLE
Addition of reference voltage calibration

### PR DESCRIPTION
**Pull request by wsnel**
Dear sabas1080,

I've been experimenting with the internal temperature sensor, using your library. Since the internal sensor is a bit noisy, I added the options to perform averaging. Additionally added the refinement calibration using the factory calibration data for the 1V internal voltage reference. While this does improve accuracy, device variability remains very visible. This means that user calibration is still a good idea. For this I added additional options as well. 
The effect of these additions are explained using an additional example sketch. I sincerely think these changes provide improvements to existing users, without breaking existing code.

Kind regards,
Walter

### CODE REVIEW

**Comment by sabas1080** [comment]
wow! Nice work @wsnel , I 'm checking

File: src/TemperatureZero.cpp [hunk 1]
```diff
@@ -7,31 +7,135 @@
...
+    Serial.print(F("Hot Voltage ref raw / interpreted  : "));
+    Serial.print(hotInt1vRefRaw);
+    Serial.print(F(" / "));
+    Serial.println(_hotInt1vRef, 4);
+    Serial.print(F("Room Reading compensated : "));
+    Serial.println(_roomVoltageCompensated, 4);
+    Serial.print(F("Hot Reading compensated  : "));
+    Serial.println(_hotVoltageCompensated, 4);
+#endif
+}
```
Comment (sabas1080) [inline src/TemperatureZero.cpp]: @wsnel debug via serial it is not the best way to do it, we recommend removing or reviewing the following article https://www.sparkfun.com/news/3245 "Don’t Serial.print in your library"

**Review by sabas1080 — CHANGES_REQUESTED** [review]

File: src/TemperatureZero.cpp (hunk 1 above)
Comment (wsnel) [reply src/TemperatureZero.cpp]: Great link, changes are in. Additionally examples are now more according to SparkFun recommendations as well.

**Review by sabas1080 — APPROVED** [review]

**Comment by sabas1080** [comment]
@wsnel great work, thanks!

**Pull request merged**